tests/lib/unit/API/Repository/Value/Filter/FilterTest.php

```diff
@@ -0,0 +1,323 @@
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function testOrWithCriterion(): Filter
+    {
+        // sanity check
+        $criterion1 = new Criterion\ContentId(1);
+        $criterion2 = new Criterion\RemoteId(1);
+        $criterion3 = new Criterion\Ancestor('/1/2/');
+
+        $filter = new Filter();
+        $filter->withCriterion($criterion1);
+
+        $filter->orWithCriterion($criterion2);
+        $expectedCriterion = new Criterion\LogicalOr([$criterion1, $criterion2]);
+        self::assertEquals($expectedCriterion, $filter->getCriterion());
+
+        $filter->orWithCriterion($criterion3);
+        $expectedCriterion = new Criterion\LogicalOr([$criterion1, $criterion2, $criterion3]);
+        self::assertEquals($expectedCriterion, $filter->getCriterion());
+
+        return $filter;
+    }
+
+    /**
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::withSortClause
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::getSortClauses
+     */
+    public function testWithSortClause(): Filter
+    {
+        $filter = new Filter();
+        // sanity check
+        self::assertSame([], $filter->getSortClauses());
+
+        $sortClause1 = new SortClause\Location\Priority(Query::SORT_DESC);
+        $filter->withSortClause($sortClause1);
+        self::assertContainsEquals($sortClause1, $filter->getSortClauses());
+
+        $sortClause2 = new SortClause\Location\Priority(Query::SORT_DESC);
+        $filter->withSortClause($sortClause2);
+        self::assertContainsEquals($sortClause2, $filter->getSortClauses());
+        self::assertSame([$sortClause1, $sortClause2], $filter->getSortClauses());
+
+        return $filter;
+    }
+
+    /**
+     * @covers  \Ibexa\Platform\API\Repository\Value\Filter\Filter::reset
+     * @covers  \Ibexa\Platform\API\Repository\Value\Filter\Filter::getCriterion
+     * @covers  \Ibexa\Platform\API\Repository\Value\Filter\Filter::getSortClauses
+     *
+     * @depends testWithCriterion
+     * @depends testAndWithCriterion
+     * @depends testOrWithCriterion
+     * @depends testWithSortClause
+     */
+    public function testReset(
+        Filter $filterWithCriterion,
+        Filter $filterAndWithCriterion,
+        Filter $filterOrWithCriterion,
+        Filter $filterWithSortClause
+    ): void {
+        // sanity check
+        $filtersWithCriteria = [
+            $filterWithCriterion,
+            $filterAndWithCriterion,
+            $filterOrWithCriterion,
+        ];
+        foreach ($filtersWithCriteria as $filter) {
+            self::assertNotEmpty($filter->getCriterion());
+        }
+        self::assertNotEmpty($filterWithSortClause->getSortClauses());
+
+        foreach (func_get_args() as $filter) {
+            $filter->reset();
+            self::assertEmpty($filter->getCriterion());
+            self::assertEmpty($filter->getSortClauses());
+        }
+    }
```

Comment:
Sorry, but this implementation is over engineered. KISS:

A. Create Filter with criterion and sort clauses (simply use constructor).
B. Call `reset()` method
C. Assert if fields has been reset

Reply:
> Sorry, but this implementation is over engineered. KISS:
> 
> A. Create Filter with criterion and sort clauses (simply use constructor).
> B. Call `reset()` method
> C. Assert if fields has been reset

What if the behavior of the constructor started to differ from the behavior of the methods? We wouldn't detect breaking change. But I could simplify this test to create a separate Filter with all the calls. It just felt kinda nice to re-use previous tests results instead of wasting more memory and op. time ;)